frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideInBackgroundable.java

```diff
@@ -0,0 +1,166 @@
+        entryToSlideIn = childEntryByName.map(e -> e.withChildren(List.empty())).getOrNull();
+        targetBranchLayout = Try.of(() -> branchLayout.slideOut(slideInOptions.getName()))
+            .onFailure(e -> Match(e).of(
+                Case($(instanceOf(EntryDoesNotExistException.class)), exceptionWithMessageHandler(
+                    format(
+                        getString(
+                            "action.GitMachete.BaseSlideInBranchBelowAction.notification.message.entry-does-not-exist"),
+                        entryToSlideIn.getName()))),
+                Case($(instanceOf(EntryIsRootException.class)), exceptionWithMessageHandler(
+                    format(getString("action.GitMachete.BaseSlideInBranchBelowAction.notification.message.entry-is-root"),
+                        entryToSlideIn.getName()))),
+                Case($(), exceptionWithMessageHandler(/* message */ null))))
+            .getOrNull();
+
+        if (targetBranchLayout == null) {
+          return;
+        }
+      }
+
+    } else {
+      entryToSlideIn = new BranchLayoutEntry(slideInOptions.getName(), /* customAnnotation */ null,
+          /* children */ List.empty());
+      targetBranchLayout = branchLayout;
+    }
+
+    var newBranchLayout = Try
+        .of(() -> targetBranchLayout.slideIn(parentName, entryToSlideIn))
+        .onFailure(e -> Match(e).of(
+            Case($(instanceOf(EntryDoesNotExistException.class)), exceptionWithMessageHandler(
+                format(
+                    getString("action.GitMachete.BaseSlideInBranchBelowAction.notification.message.entry-does-not-exist"),
+                    parentName))),
+            Case($(instanceOf(EntryIsDescendantOfException.class)), exceptionWithMessageHandler(
+                format(
+                    getString("action.GitMachete.BaseSlideInBranchBelowAction.notification.message.entry-is-descendant-of"),
+                    entryToSlideIn.getName(), parentName))),
+            Case($(), exceptionWithMessageHandler(/* message */ null))))
+        .toOption();
+
+    newBranchLayout.map(nbl -> Try.run(() -> branchLayoutWriter.write(macheteFilePath, nbl, /* backupOldLayout */ true))
+        .onFailure(t -> notifier.notifyError(
+            /* title */ getString(
+                "action.GitMachete.BaseSlideInBranchBelowAction.notification.title.branch-layout-write-fail"),
+            getMessageOrEmpty(t))));
+  }
+
+  @SuppressWarnings("regexp") // needed to use forbidden "synchronized"
+  private void waitForLocalBranch() {
+    Supplier<@Nullable GitLocalBranch> findLocalBranch = () -> gitRepository.getBranches()
+        .findLocalBranch(slideInOptions.getName());
+
+    try {
+      //  6 attempts, usually 3 are enough
+      final int TIMEOUT = 2048;
+      long WAIT_DURATION = 64;
+      while (findLocalBranch.get() == null && WAIT_DURATION <= TIMEOUT) {
+        synchronized (this) {
+          wait(WAIT_DURATION);
+        }
+        WAIT_DURATION *= 2;
```

Comment:
What are the benefits of exponential waiting time here?
Couldn't it be just const time - then it can be simplified (e.g. using Java's [Timer](https://docs.oracle.com/javase/7/docs/api/java/util/Timer.html) class)

Reply:
Yeah, can be both const and exponential here, difference is gonna be rather minuscule anyway ;)

Reply:
Hm `Timer` would introduce yet another thread on which the stuff would execute... which I'd rather avoid here unless absolutely needed :/

Reply:
(since each `Timer` has an associated thread... also, I'd rather avoid using non-IDE-managed threads - so in practice, only UI thread or background tasks look like a viable option)